src/components/menu/menu.ts

```diff
@@ -240,30 +254,52 @@ export class Menu extends Component<Props, SpreadsheetChildEnv> {
     return subMenu.parentMenu?.id === menuItem.id;
   }
 
-  closeSubMenu() {
+  private closeSubMenu() {
+    if (this.subMenu.isHoveringChild) {
+      return;
+    }
     this.subMenu.isOpen = false;
     this.subMenu.parentMenu = undefined;
   }
 
-  onClickMenu(menu: Action, menuIndex: number, ev: MouseEvent) {
+  onClickMenu(menu: Action, ev: MouseEvent) {
     if (this.isEnabled(menu)) {
       if (this.isRoot(menu)) {
-        this.openSubMenu(menu, menuIndex, ev);
+        this.openSubMenu(menu, ev.currentTarget as HTMLElement);
       } else {
         this.activateMenu(menu);
       }
     }
   }
 
-  onMouseOver(menu: Action, position: Pixel, ev: MouseEvent) {
+  onMouseOver(menu: Action, ev: MouseEvent) {
     if (this.isEnabled(menu)) {
+      if (this.isParentMenu(this.subMenu, menu)) {
+        this.openingTimeOut.clear();
+        return;
+      }
+      const currentTarget = ev.currentTarget as HTMLElement;
       if (this.isRoot(menu)) {
-        this.openSubMenu(menu, position, ev);
-      } else {
-        this.closeSubMenu();
+        this.openingTimeOut.schedule(() => {
+          this.openSubMenu(menu, currentTarget);
+        }, TIMEOUT_DELAY);
       }
     }
   }
+
+  onMouseOverMainDiv(ev: MouseEvent) {
```

Comment:
The reference to 'div' in the mouseover functions feels too technical. since we are handling menus, could we considre MainMenu and ChildMenu?